include/execution.hpp

```diff
@@ -2461,7 +2461,109 @@ namespace _P2300::execution {
+        }
+
+        template <class... _As>
+        void set_value(_As&&... __as) && noexcept 
+          requires __callable<_Fun, _Shape, _As&...> {
+          try {
+            for (_Shape __i{}; __i != __shape_; ++__i) {
+              __f_(__i, __as...);
+            }
+            execution::set_value(std::move(this->base()), (_As&&)__as...);
+          } catch(...) {
+            execution::set_error(std::move(this->base()), std::current_exception());
+          }
+        }
+
+       public:
+        explicit __receiver(_Receiver __rcvr, _Shape __shape, _Fun __fun)
+          : receiver_adaptor<__receiver, _Receiver>((_Receiver&&) __rcvr)
+          , __shape_(__shape)
+          , __f_((_Fun&&) __fun)
+        {}
+      };
+
+    template <class _SenderId, integral _Shape, class _FunId>
+      struct __sender {
+        using _Sender = __t<_SenderId>;
+        using _Fun = __t<_FunId>;
+        template <receiver _Receiver>
+          using __receiver = __receiver<__x<remove_cvref_t<_Receiver>>, _Shape, _FunId>;
+
+        [[no_unique_address]] _Sender __sndr_;
+        [[no_unique_address]] _Shape __shape_;
+        [[no_unique_address]] _Fun __fun_;
+
+        template <class _Fun, class _Sender, class _Env>
+          using __with_error_invoke_t =
+            __if_c<
+              __v<__value_types_of_t<
+                _Sender,
+                _Env,
+                __mbind_front_q<__non_throwing_, _Fun, _Shape>,
+                __q<__mand>>>,
+              completion_signatures<>,
+              __with_exception_ptr>;
+
+        template <class _Self, class _Env>
+          using __completion_signatures =
+            __make_completion_signatures<
+              __member_t<_Self, _Sender>,
+              _Env,
+              __with_error_invoke_t<_Fun, __member_t<_Self, _Sender>, _Env>>;
+
+        template <__decays_to<__sender> _Self, receiver _Receiver>
+          requires sender_to<__member_t<_Self, _Sender>, __receiver<_Receiver>>
+        friend auto tag_invoke(connect_t, _Self&& __self, _Receiver&& __rcvr)
+          noexcept(__nothrow_connectable<__member_t<_Self, _Sender>, __receiver<_Receiver>>)
+          -> connect_result_t<__member_t<_Self, _Sender>, __receiver<_Receiver>> {
+          return execution::connect(
+              ((_Self&&) __self).__sndr_,
+              __receiver<_Receiver>{
+                (_Receiver&&) __rcvr, 
+                __self.__shape_, 
+                ((_Self&&) __self).__fun_});
+        }
+
+        template <__decays_to<__sender> _Self, class _Env>
+        friend auto tag_invoke(get_completion_signatures_t, _Self&&, _Env)
+          -> dependent_completion_signatures<_Env>;
+
+        template <__decays_to<__sender> _Self, class _Env>
+        friend auto tag_invoke(get_completion_signatures_t, _Self&&, _Env)
+          -> __completion_signatures<_Self, _Env> requires true;
+
+        template <__sender_queries::__sender_query _Tag, class... _As>
```

Comment:
Seems like something might be missing here?  I don't see __sender_query having ever been defined.

```
$ git grep __sender_query
execution.hpp:    template <__sender_queries::__sender_query _Tag, class... _As>
execution.hpp:            -> __call_result_if_t<__sender_queries::__sender_query<_Tag>, _Tag,
```

On GCC I'm seeing:
```
../include/sdbusplus/async/std_execution/execution.hpp:3001:15: error: ‘_P2300::execution::__sender_queries::__sender_query’ has not been declared
 3001 |     template <__sender_queries::__sender_query _Tag, class... _As>
```

Reply:
@williamspatrick I believe it was removed in the following [PR](https://github.com/brycelelbach/wg21_p2300_std_execution/pull/581). It should be `tag_category<forwarding_sender_query>` now. I fixed that locally and will create a PR soon.